x-pack/plugins/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -142,18 +213,87 @@ export function Insight({ messages, title, dataTestSubj }: InsightProps) {
     [service]
   );
 
+  const handleSend = (newPrompt: string) => {
+    const clonedMessages = cloneDeep(messages);
+    const message = last(clonedMessages.filter((msg) => msg.message.role === MessageRole.User));
+    if (!message) return false;
+
+    message.message.content = newPrompt;
+    setUpdatedPrompt(newPrompt);
+    setInitialMessages(clonedMessages);
+    setEditingPrompt(false);
+    return true;
+  };
+
+  const handleCancel = () => {
+    setEditingPrompt(false);
+    setInsightOpen(false);
+    setHasOpened(false);
+  };
+
   const {
     services: { http },
   } = useKibana();
 
   let children: React.ReactNode = null;
 
-  if (hasOpened && connectors.selectedConnector) {
+  if (
+    connectors.selectedConnector &&
+    ((!isInsightOpen && hasOpened) || (isInsightOpen && !isEditingPrompt))
+  ) {
     children = (
-      <ChatContent
-        title={title}
-        initialMessages={messages}
-        connectorId={connectors.selectedConnector}
+      <>
+        {updatedPrompt ? (
+          <>
+            <EuiFlexGroup alignItems="center" gutterSize="none">
+              <EuiFlexItem grow={false}>
+                <EuiText size="xs">
+                  {i18n.translate('xpack.observabilityAiAssistant.insightModifiedPrompt', {
+                    defaultMessage: 'This insight has been modified.',
+                  })}
+                </EuiText>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  data-test-subj="observabilityAiAssistantInsightResetDefaultPrompt"
+                  onClick={() => {
+                    setUpdatedPrompt('');
+                    setHasOpened(false);
+                    setInsightOpen(false);
+                    setInitialMessages(messages);
+                  }}
+                >
+                  <EuiText size="xs">
+                    {i18n.translate('xpack.observabilityAiAssistant.resetDefaultPrompt', {
+                      defaultMessage: 'Reset to default',
+                    })}
+                  </EuiText>
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+
+            <EuiHorizontalRule size="full" margin="none" />
+            <EuiSpacer size="m" />
+          </>
+        ) : null}
+
+        <ChatContent
+          title={title}
+          initialMessages={initialMessages}
+          connectorId={connectors.selectedConnector}
+        />
+      </>
+    );
+  } else if (isEditingPrompt) {
+    children = (
+      <PromptEdit
+        initialPrompt={
+          updatedPrompt ||
+          last(messages.filter((msg) => msg.message.role === MessageRole.User))?.message.content ||
+          ''
+        }
```

Comment:
Do you need `updatedPrompt`? You already update `messages` in `handleSend` so I'd assume `updatedPrompt` is identical to `last(messages.filter((msg) => msg.message.role === MessageRole.User))?.message.content`.

If you don't need `updatedPrompt` here it should be a boolean `didChangePrompt` or something like that. 

```suggestion
        initialPrompt={
          last(messages.filter((msg) => msg.message.role === MessageRole.User))?.message.content ||
          ''
        }
```

Reply:
good catch, updated in https://github.com/elastic/kibana/pull/176086/commits/fa7d3601e5da7a355db4969da56fa305708c5987